symengine/dict.h

```diff
@@ -49,20 +49,30 @@ typedef std::map<unsigned, integer_class> map_uint_mpz;
 typedef std::map<unsigned, rational_class> map_uint_mpq;
 typedef std::map<int, Expression> map_int_Expr;
 typedef std::vector<integer_class> vec_integer_class;
-//! Part of umap_vec_mpz:
-typedef struct {
-    inline std::size_t operator()(const vec_int &k) const
+typedef std::vector<unsigned int> vec_uint;
+typedef std::unordered_map<RCP<const Basic>, unsigned int, RCPBasicHash,
+                           RCPBasicKeyEq> umap_basic_uint;
+
+template <typename T>
+class vec_hash
+{
+public:
+    std::size_t operator()(const T &v) const
     {
         std::size_t h = 0;
-        for (const auto &p : k) {
-            h = (h << 4) + p;
+        for (auto i : v) {
+            h ^= i + 0x9e3779b + (h << 6) + (h >> 2);
```

Comment:
Initially `vector<uint>` used the above hash function, while `vector<int>` used the previous one.
I thought it would make sense to use the same hash functions for both. I also remember Isuru saying that the (older) hash function for `vector<int>` wasn't very good.